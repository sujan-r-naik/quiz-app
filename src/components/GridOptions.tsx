import {
  Grid,
  Typography,
  TextField,
  Button,
  InputAdornment,
} from "@mui/material";
import React from "react";

interface IOptions {
  ans: any;
  options: any;
  ansIndex: number;
  clickedAns: Function;
  currIndex: number;
  mapIndex: number;
  setMatchData: React.Dispatch<
    React.SetStateAction<{
      1: string;
      2: string;
      3: string;
      4: string;
    }>
  >;
}

const GridOptions = (props: IOptions) => {
  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    // console.log({ [parseInt(e.target.name)]: e.target.value });
    props.setMatchData((prev) => ({
      ...prev,
      [parseInt(e.target.name)]: e.target.value,
    }));
  };
  return (
    <>
      {props.currIndex === 4 ? (
        <Grid item xs={5} role="gridOpt4">
          <TextField
            inputProps={{ "data-testid": "match-inp", maxLength: 1 }}
            id="outlined-start-adornment"
            sx={{ m: 1, width: "25ch" }}
            name={`${props.mapIndex + 1}`}
            onChange={handleChange}
            InputProps={{
              startAdornment: (
                <InputAdornment position="start">
                  {props.mapIndex + 1}:
                </InputAdornment>
              ),
            }}
          />
        </Grid>
      ) : (
        <Grid
          role={`gridOpt-${props.mapIndex}`}
          item
          xs={5}
          onClick={() => {
            props.clickedAns(props.ansIndex);
          }}
        >
          <Button
            role="optBtn"
            variant="outlined"
            sx={{
              width: "100%",
              p: 1,
              backgroundColor:
                props.ans !== {} && props.ans.answerText === props.options
                  ? "lightskyblue"
                  : "none",
              borderRadius: 1,
              "&:active": {
                backgroundColor: "lightskyblue",
              },
              "&:focus": {
                backgroundColor: "powderblue",
              },
            }}
          >
            <Typography sx={{ textAlign: "center" }} role="optopt">
              {props.options}
            </Typography>
          </Button>
        </Grid>
      )}
    </>
  );
};

export default GridOptions;
