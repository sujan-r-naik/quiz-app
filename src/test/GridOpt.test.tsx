import "@testing-library/jest-dom/extend-expect";
import { screen, render, fireEvent } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import GridOptions from "../components/GridOptions";
import renderer from "react-test-renderer";
import userEvent from "@testing-library/user-event";

test("Snapshot of NumberNav component", () => {
  const comp = renderer.create(
    <BrowserRouter>
      <GridOptions
        ans={{}}
        options={"A"}
        ansIndex={0}
        clickedAns={() => {}}
        currIndex={1}
        mapIndex={1}
        setMatchData={() => {}}
      />
    </BrowserRouter>
  );
  let tree = comp.toJSON();
  expect(tree).toMatchSnapshot();
});

describe("In Quiz,", () => {
  test("does the sub components of gridOptions work at index !=4", () => {
    render(
      <BrowserRouter>
        <GridOptions
          ans={{}}
          options={"A"}
          ansIndex={0}
          clickedAns={() => {}}
          currIndex={0}
          mapIndex={1}
          setMatchData={() => {}}
        />
      </BrowserRouter>
    );

    let a = screen.getByRole("gridOpt-1");
    let b = screen.getByRole("optopt");
    let c = screen.getByRole("optBtn");
    expect(a).toBeInTheDocument();
    expect(b).toBeInTheDocument();
    expect(c).toBeInTheDocument();
  });

  test("does the sub components of gridOptions work at index ===4", () => {
    render(
      <BrowserRouter>
        <GridOptions
          ans={{}}
          options={"A"}
          ansIndex={0}
          clickedAns={() => {}}
          currIndex={4}
          mapIndex={3}
          setMatchData={() => {}}
        />
      </BrowserRouter>
    );

    let a = screen.getByRole("gridOpt4");
    let matchinp = screen.getByTestId("match-inp") as HTMLInputElement;
    expect(a).toBeInTheDocument();
    expect(matchinp).toBeInTheDocument();

    userEvent.type(matchinp, "K");
    expect(screen.getByRole("textbox")).toHaveValue("Hello,\nWorld!");
  });
});
