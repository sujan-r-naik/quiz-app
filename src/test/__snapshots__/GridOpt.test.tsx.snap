// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of NumberNav component 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-13i4rnv-MuiGrid-root"
  onClick={[Function]}
  role="gridOpt-1"
>
  <button
    className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1ar33r0-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="optBtn"
    tabIndex={0}
    type="button"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-1uwgr7b-MuiTypography-root"
      role="optopt"
    >
      A
    </p>
  </button>
</div>
`;
